code/drasil-code/lib/Language/Drasil/Code/Imperative/ReadInput.hs

```diff
@@ -98,12 +98,12 @@ splitAtFirst = splitAtFirst' []
         dropDelim [] s = s
         dropDelim _ [] = error "impossible"
 
--- | Converts a list of 'String's to a Matrix 'Expr' of a given 'Space'.
+-- | Converts a list of 'String's to a Clif 'Expr' of a given 'Space'.
 strListAsExpr :: Space -> [String] -> Expr
-strListAsExpr (Vect t) ss = Matrix [map (strAsExpr t) ss]
+strListAsExpr (ClifS gr d s) ss = undefined -- TODO: fill this in
 strListAsExpr _ _ = error "strListsAsExpr called on non-vector space"
 
--- | Converts a 2D list of 'String's to a Matrix 'Expr' of a given 'Space'.
+-- | Converts a 2D list of 'String's to a Clif 'Expr' of a given 'Space'.
 strList2DAsExpr :: Space -> [[String]] -> Expr
-strList2DAsExpr (Vect (Vect t)) sss = Matrix $ map (map (strAsExpr t)) sss
+strList2DAsExpr (ClifS gr0 d0 (ClifS gr1 d1 s)) sss = undefined -- TODO: fill this in
```

Comment:
Trying to fill these particular TODOs will be very instructive.

Reply:
This is another question I had: does geometric algebra have such a thing as "clifs of clifs"?

Reply:
No, GA does not nest.

Reply:
Didn't think so. So, if Clifs take `Expr` as an argument, we will have to have type checking to make sure they don't take another clif as an argument.

Reply:
Correct.